pkg/deploy/generator/resources.go

```diff
@@ -1089,6 +1052,124 @@ EOF
 
 chcon -R system_u:object_r:var_log_t:s0 /var/opt/microsoft/linuxmonagent
 
+cat >/etc/systemd/system/download-mdm-credentials.service <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/download-credentials.sh mdm
+EOF
+
+cat >/etc/systemd/system/download-mdm-credentials.timer <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Timer]
+OnBootSec=0min
+OnCalendar=0/12:00:00
+AccuracySec=5s
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/local/bin/download-credentials.sh <<EOF
+#!/bin/bash
+set -eu
+
+COMPONENT="\$1"
+if [[ "\$COMPONENT" = "mdm" || "\$COMPONENT" = "mdsd" ]]; then
+  NEW_CERT_FILE="/etc/temp-\${COMPONENT}.pem"
+  CURRENT_CERT_FILE="/etc/\${COMPONENT}.pem"
```

Comment:
In mdsd's case, the file is going to need to end up under /var/lib/waagent/Microsoft.Azure.KeyVault.Store unless you override the MDSD_AKV_CERTIFICATE_STORE_PATH environment variable.
If you override the environment variable (possibly a better approach), the file will need to be placed in a directory which has no other certificates.

Reply:
I am aware of this. This and some other minor clean up are the reason I kept this in Draft still https://github.com/Azure/ARO-RP/pull/1240#issuecomment-759990105